### ISSUE TITLE
Git changes not show in the file explorer

### ISSUE THREAD

**Issue by MebDevelop** [microsoft/vscode#89104]
- VSCode Version:
- OS Version:
Version: 1.41.1
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2019-12-18T14:57:51.166Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 17.7.0

Steps to Reproduce:

1. I can't see the git changes in a specific folder
![file explorer](https://i.ibb.co/cbcTQSk/file-explorer.png)

2. But I can see the changes in source control
![source control](https://i.ibb.co/3h5BJdF/source-control.png)

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by MebDevelop** [comment]
I am still having issue. Please fix this issue asap.

**Comment by joaomoreno** [comment]
Seems same as https://github.com/microsoft/vscode/issues/89143.

Can you send me the full output of `Git: Show Git Output`?

**Comment by joaomoreno** [comment]
Interesting that `index.js` is colored as if there should be a decoration. Do you have horizontal scrolling enabled?

**Comment by MebDevelop** [comment]
`Git: Show Git Output` is not working.
This is output is of git show.
![img](https://i.ibb.co/qMt5bHg/Screen-Shot-2020-01-23-at-4-04-58-AM.png)

**Comment by joaomoreno** [comment]
What do you mean by _is not working_? Did you run it? What happened when you did? Was there an error?

**Comment by MebDevelop** [comment]
![error](https://i.imgur.com/l4pZpwG.png)

**Comment by MebDevelop** [comment]
Well "M" is showing on outside of 'src' folder of my project. 
For example It is showing in the gitignore but I can't see 'M' in the files of src folder.

**Comment by MebDevelop** [comment]
no update?

**Comment by joaomoreno** [comment]
`Git: Show Git Output` is a VS Code command. Press F1 and run it there.

**Comment by MebDevelop** [comment]
Looking for git in: /usr/bin/git
Using git 2.17.2 (Apple Git-113) from /usr/bin/git
> git rev-parse --show-toplevel
> git rev-parse --git-dir
Open repository: /Volumes/DATA/Work/Mobile/ReactNative/businessCardScanner
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse master
> git rev-parse --symbolic-full-name master@{u}
> git rev-list --left-right master...refs/remotes/origin/master
> git show :src/Pages/SettingsView/AppSettings.js
> git check-ignore -v -z --stdin
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git config --get commit.template
> git check-ignore -v -z --stdin
> git check-ignore -v -z --stdin
> git check-ignore -v -z --stdin
> git check-ignore -v -z --stdin
> git show :src/Utils/Localization/German.json
> git show :src/Utils/Localization/en.json
> git check-ignore -v -z --stdin

**Comment by joaomoreno** [comment]
Do you have horizontal scrolling enabled?

**Comment by MebDevelop** [comment]
Did you mean this one?
![img](https://i.imgur.com/K56Rfz1.png)

**Comment by joaomoreno** [comment]
Yes, I'm guessing you don't. Do you mind sharing your full user settings with me?

**Comment by MebDevelop** [comment]
Sure, I can. How can I share my full user settings?

**Comment by MebDevelop** [comment]
{
  "editor.formatOnSave": false,
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  },
  "eslint.validate": [
    "javascript",
    "javascriptreact",
  ]
}

It's my custom settings.json file.

**Comment by joaomoreno** [comment]
Can you reproduce this in the [insider release](https://code.visualstudio.com/insiders)?

Can you reproduce in another repositories?

**Comment by MebDevelop** [comment]
Well, I can't with another repository

**Comment by MebDevelop** [comment]
I can reproduce with insider release. the weird thing is I can't see it in only src folder, but I can see in android and ios and some other files in the top level like package.json.

**Comment by joaomoreno** [comment]
What's your git version?

**Comment by MebDevelop** [comment]
git version 2.17.2 (Apple Git-113)

**Comment by joaomoreno** [comment]
> I can reproduce with insider release. the weird thing is I can't see it in only src folder, but I can see in android and ios and some other files in the top level like package.json.

Cna you share a large screenshot of the explorer showing this exactly?

**Comment by MebDevelop** [comment]
![full image](https://i.imgur.com/jIjJlam.png)

**Comment by MebDevelop** [comment]
As you can see in there I can't find any highlighted icon in the subfolders of src folder even I have edited file in the Utils folder

**Comment by MebDevelop** [comment]
But I can see the edited file('M') in android folder as well.

**Comment by joaomoreno** [comment]
If you clone the same repository into another folder, can you reproduce? If so, can you share the repository with us?

**Comment by MebDevelop** [comment]
I think it's working in another folder.

**Comment by joaomoreno** [comment]
Can you please confirm? Same repo in another folder.

**Comment by MebDevelop** [comment]
Yes, I confirmed. It was working

**Comment by joaomoreno** [comment]
I mean, same computer, same repository, different checkout folder.

**Comment by MebDevelop** [comment]
It's working on same computer, same repository, different checkout folder

**Comment by MebDevelop** [comment]
Do you want to let me move to work on another folder?

**Comment by joaomoreno** [comment]
If that solves it, yeah! :+1:

**Comment by MebDevelop** [comment]
But I want to work on previous folder. And I think you should solve that problem for the vscode, don't u?

**Comment by joaomoreno** [comment]
Can you share the exact locations of both paths? So we figure out what's the difference between them?

**Comment by MebDevelop** [comment]
The previous one is in /Volumes/DATA/Work/Mobile/ReactNative/businessCardScanner
The working one is in /Volumes/DATA/Work/vcard

**Comment by joaomoreno** [comment]
Let's try another experiment: can you backup `/Volumes/DATA/Work/Mobile/ReactNative/businessCardScanner` and then move `vcard` to exactly that location `/Volumes/DATA/Work/Mobile/ReactNative/businessCardScanner`?

**Comment by MebDevelop** [comment]
Well, I move previous one to another folder and move working one to businessCardScanner and it's working.

**Comment by joaomoreno** [comment]
OK there must be something inside that original folder which is blocking the feature. Unless you send me the entire folder, there's no way I can find out what it is. Since you are unblocked in more than one way, I'd close this for now. Let me know if you eventually find what is it about that folder that prevents the feature from working.

**Comment by PrzemyslawKlys** [comment]
I actually have the same problem. It started happening recently for me.

**Comment by MebDevelop** [comment]
Well, are you having same issue with me? Can't reproduce error in another folder?

**Comment by PrzemyslawKlys** [comment]
![image](https://user-images.githubusercontent.com/15063294/73296915-e2c6b180-420a-11ea-9c38-7d9d47d5cf58.png)

![image](https://user-images.githubusercontent.com/15063294/73296923-e823fc00-420a-11ea-9b41-de57c31dacad.png)

It randomly works from project to project. I have no clue why suddenly some projects would show up properly and some don't.

**Comment by MebDevelop** [comment]
Well, I am using mac and I didn't change the git version. Also, I can see the changes in the project explorer after I clone repo in another folder.

**Comment by PrzemyslawKlys** [comment]
Weird :/

**Comment by MebDevelop** [comment]
Yes, it's too weird. @joaomoreno wants to check my code to fix the issue, but I can't share code since it's our company's project.  Please update if you could find any issue.

**Comment by PrzemyslawKlys** [comment]
I have no problem to share my projects as those are open source anyways.

**Comment by PrzemyslawKlys** [comment]
I just copied the folder to Desktop and it works there. Properly. Are you using symlinks/hardlinks by any chance?

**Comment by PrzemyslawKlys** [comment]
I store my folders in C:\Support\Github\<projects>

When I move it to Desktop it works, when I move the folder to C:\Support\Github\OtherProject\<ThisProject> it doesn't work. So clearly something related to C:\Support\GitHub is broken.

What I did in Windows: `mklink /H /J "C:\Support" "C:\Users\pklys\OneDrive - Evotec\Support"`

This links my OneDrive folder to C:\Support.

![image](https://user-images.githubusercontent.com/15063294/73300134-c75ea500-4210-11ea-80ce-c1519b2b19a6.png)

It may be related.

**Comment by MebDevelop** [comment]
Well, I am not sure what is symlinks/hardlink. Do you know the command in mac?

**Comment by PrzemyslawKlys** [comment]
'ln -s' which is symbolic link
'ln' which is hardlink

Basically you can tell your drive that you want same file in multiple places and it doesn't duplicate data.

**Comment by joaomoreno** [comment]
@PrzemyslawKlys You have https://github.com/microsoft/vscode/issues/89373

**Comment by PrzemyslawKlys** [comment]
Yep, downgrade fixed it. Thanks.

**Comment by MebDevelop** [comment]
@PrzemyslawKlys, I am not sure how can I use that 'ln' command. 
As you know, I am using macOS not windows and git version is 2.17.2
What should I have to do? Downgrade git version?

**Comment by PrzemyslawKlys** [comment]
If you don't know how to use LN it means you are not using it so it's not affecting you. In my case it was git problem and after downgrade it now works properly. Not sure about mac. You would have to check when was the last time the version upgrade happend for you.

**Comment by MebDevelop** [comment]
I have tried to install several git version but it's still not working.

**Comment by MebDevelop** [comment]
I think I found the issue, before, I changed my folder names.
Originally there were src/pages but I changed it to src/Pages. Renaming to pages working for me.
Thank you for your time and help.

**Comment by noadek** [comment]
I had the same issue, had to use the terminal until I found the fix. Turns out one of the extensions was initialising or trying to but couldn't. In my case it was the angular language service extension and since I was working on a react project, I disabled this for the workspace (from extensions, click on the gear button then select Disable (Workspace)) and after the reload Git explorer was back.

Maybe this could help someone.
